### PR TITLE
Ibtida testnet chain spec

### PR DESCRIPTION
**Pull request by justinfrevert**
Changes:
- Sets token decimals
- Sets address prefix for Ibtida
- Adds a JSON chain spec for the `ibtida` testnet

The main goal of this is to get some code together to run two `ibtida` nodes. Some of this process will be manual, until we figure out some automation that makes sense for us. Given this, some pieces are intentionally omitted from this PR for containing secrets, etc.

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -156,22 +213,20 @@ fn testnet_genesis(
 			changes_trie_config: Default::default(),
 		}),
 		pallet_balances: Some(BalancesConfig {
-			// Configure endowed accounts with initial balance of 1 << 60.
-			balances: endowed_accounts
+			balances: founder_allocation
 				.iter()
-				.cloned()
-				.map(|k| (k, 1 << 60))
+				.map(|x| (x.0.clone(), x.1.clone()))
```
Comment (aelesbao) [inline node/src/chain_spec.rs]: You can use a Tuple destructuring here:

```suggestion
				.map(|(accountId, balance)| (accountId.clone(), balance.clone()))
```

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -156,22 +213,20 @@ fn testnet_genesis(
...
-			authorities: initial_authorities.iter().map(|x| (x.0.clone())).collect(),
-		}),
 		pallet_grandpa: Some(GrandpaConfig {
 			authorities: initial_authorities
 				.iter()
 				.map(|x| (x.1.clone(), 1))
 				.collect(),
 		}),
+		pallet_aura: Some(AuraConfig {
+			authorities: initial_authorities.iter().map(|x| (x.0.clone())).collect(),
```
Comment (aelesbao) [inline node/src/chain_spec.rs]: Same here:

```suggestion
			authorities: initial_authorities.iter().map(|(auraId, _)| (auraId.clone())).collect(),
```

**Review by aelesbao — DISMISSED** [review]

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -181,3 +239,40 @@ fn testnet_genesis(
 		}),
 	}
 }
+
+// testnet config
+pub fn ibtida_config() -> Result<ChainSpec, String> {
```
Comment (acgxv) [inline node/src/chain_spec.rs:241]: should not boot from here

File: node/src/command.rs [hunk 4]
```diff
@@ -49,6 +49,7 @@ impl SubstrateCli for Cli {
 	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
 		Ok(match id {
 			"dev" => Box::new(chain_spec::development_config()?),
+			"ibtida" => Box::new(chain_spec::ibtida_config()?),
```
Comment (acgxv) [inline node/src/command.rs]: should not boot from here

**Comment by aurexav** [comment]
https://github.com/paritytech/polkadot/blob/master/node/service/src/chain_spec.rs#L127-L146

File: node/src/chain_spec.rs (hunk 3 above)
Comment (justinFrevert) [reply node/src/chain_spec.rs:244]: Probably worth just removing. No problem keeping the JSON chainspec, right?

Comment (nblogist) [reply node/src/chain_spec.rs:241]: @justinFrevert ?

**Review by aelesbao — APPROVED** [review]

**Comment by justinfrevert** [comment]
We should follow-on and point the `ibtida` command argument at the new raw chainspec path as Xavier mentioned

**Pull request merged**